lib/src/over_react_test/custom_matchers.dart

```diff
@@ -337,3 +339,139 @@ Matcher throwsPropError_Combination(String propName, String prop2Name, [String m
       ))
   ));
 }
+
+/// A log matcher that captures logs and uses a provided matcher or string to compare
+/// against.
+///
+/// The primary use case of the matcher is to take in a callback as the actual,
+/// and pass it to [recordConsoleLogs] to run the function and record the resulting
+/// logs that are emitted during the function runtime.
+class _LoggingFunctionMatcher extends CustomMatcher {
+  /// [matcher] is tested against the list of recorded logs
+  // could make wrapMatcher be `contains` by default as well if desired
+  _LoggingFunctionMatcher(dynamic matcher, {this.config, description, name})
+      : super(description ?? 'emits the logs', name ?? 'logs', _customWrapMatcher(matcher));
+
+  final ConsoleConfiguration config;
+
+  static Matcher _customWrapMatcher(dynamic x) {
+    if (x == null || (x is List && x.isEmpty)) {
+      throw ArgumentError('The expected value cannot be null or an empty list. \n\n'
+          'If you are expecting no logs, use the emitsNoLogs matcher');
+    }
+
+    if (x is Matcher) {
+      return x;
+    } else if (x is String) {
+      return anyElement(contains(x));
+    } else {
+      return equals(x);
+    }
+  }
+
+  @override
+  featureValueOf(actual) {
+    var logs = <String>[];
+
+    if (actual is List) return actual;
+
+    if (actual is! Function()) {
+      throw ArgumentError('The actual value must be a callback or a List.');
+    }
+
+    try {
+      logs = recordConsoleLogs(actual, config ?? logConfig);
+    } catch (_) {}
+
+    return logs;
+  }
+}
+
+/// A Matcher used to compare a list of logs against a provided matcher.
+///
+/// __Examples:__
+///
+///     When passed a string, the matcher look for any log that contains that
+///     substring.
+///     ```dart
+///       expect(callbackFunction, emitsLogs('I expect this log'));
```

Comment:
I feel like it emitsLogs handling both single and multiple log expectations could get confusing, since it's not immediately clear when and how the actual expectations are converted to their corresponding matchers.

What are your thoughts on having an `emitsLog` for that single case and only supporting Iterable matchers for `emitsLogs`?